firedrake/preconditioners/pmg.py

```diff
@@ -162,15 +175,15 @@ def coarsen(self, fdm, comm):
         # Coarsen the _SNESContext of a DM fdm
         # return the coarse DM cdm of the coarse _SNESContext
         fctx = get_appctx(fdm)
+        parent = get_parent(fdm)
+        assert parent is not None
 
         # Have we already done this?
+        # FIXME this is triggering weird gmg erros
```

Comment:
Is this comment still relevant? It sounds like it should be addressed before merge.

Reply:
This was giving me errors for different coarsening schedules within a nested iteration (p-FAS/p-MG). I resolved this by ensuring that the coarse `_SNESContext` cached by the outer solver is in fact the one required by the inner solver. When this is not the case, we need to rediscretize the problem with the degree of the inner solver and overwrite the `_coarse` attribute of the `_SNESContentext`.